debug_module.tex

```diff
@@ -229,13 +217,32 @@ \section{Abstract Commands} \label{abstractcommands}
     \end{tabulary}
 \end{table}
 
+\begin{commentary}
+    The Abstract Command interface is designed to allow a debugger to write
+    commands as fast as possible, and then later check whether they completed
+    without error.  In the common case the debugger will be much slower than
+    the target and commands succeed, which allows for maximum throughput. If
+    there is a failure, the interface ensures that no commands execute after
+    the failing one.  To discover which command failed, the debugger has to
+    look at the state of the DM (e.g.\ contents of \Rdatazero) or hart (e.g.\ 
+    contents of a register modified by a Program Buffer program) to determine
+    which one failed.
+\end{commentary}
+
 Before starting an abstract command, a debugger must ensure that \Fhaltreq,
 \Fresumereq, and \Fackhavereset are all 0.
 
 While an abstract command is executing (\Fbusy in \Rabstractcs is high), a
 debugger must not change \Fhartsel, and must not write 1 to \Fhaltreq,
 \Fresumereq, \Fackhavereset, \Fsetresethaltreq, or \Fclrresethaltreq.
 
+If a hart is reset while executing an abstract command then the abstract
```

Comment:
with regards to the discussion in #354 , should this be "if a hart becomes unavailable while executing an abstract command"?

I'll say it again: we're mixing two things here: what happens to an abstract command if a hart becomes unavailable? And how can I cancel a running abstract command (For whatever reason)?

You can already cancel a running abstract command by resetting the debug module (via DMACTIVE). So let's not mix the desire to have a way to do that with the potential error case of what happens when a hart becomes unavailable/reset during the command which is executing.

Reply:
Yes, it will have to be "if a hart becomes unavailable...."